tests/meshcop/joiner_session.hpp

```diff
@@ -0,0 +1,144 @@
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   The file is the header for the dtls session class to communicate with the joiner
+ */
+
+#ifndef OTBR_JOINER_SESSION_HPP_
+#define OTBR_JOINER_SESSION_HPP_
+
+#include <arpa/inet.h>
+#include <errno.h>
+#include <sys/socket.h>
+
+#include "commissioner_constants.hpp"
+#include "agent/coap.hpp"
+#include "agent/dtls.hpp"
+#include "web/pskc-generator/pskc.hpp"
+
+namespace ot {
+namespace BorderRouter {
+
+class JoinerSession
+{
+public:
+    /**
+     * The constructor to initialize JoinerSession
+     *
+     * @param[in]    aInternalServerPort    port for internal dtls server to listen to
+     * @param[in]    aPskdAscii             ascii form of pskd
+     *
+     */
+    JoinerSession(uint16_t aInternalServerPort, const char *aPskdAscii);
+
+    /**
+     * This method updates the fd_set and timeout @p aTimeout should
+     * only be updated if session has pending process in less than its current value.
+     *
+     * @param[inout]    aReadFdSet      A reference to fd_set for polling read.
+     * @param[inout]    aWriteFdSet     A reference to fd_set for polling write.
+     * @param[inout]    aErrorFdSet     A reference to fd_set for polling error.
+     * @param[inout]    aMaxFd          A reference to the current max fd in @p aReadFdSet and @p aWriteFdSet.
+     * @param[inout]    aTimeout        A reference to the timeout.
+     *
+     */
+    void UpdateFdSet(fd_set &aReadFdSet, fd_set &aWriteFdSet, fd_set &aErrorFdSet, int &aMaxFd, timeval &aTimeout);
+
+    /**
+     * This method performs the session processing.
+     *
+     * @param[in]   aReadFdSet          A reference to fd_set ready for reading.
+     * @param[in]   aWriteFdSet         A reference to fd_set ready for writing.
+     * @param[in]   aErrorFdSet         A reference to fd_set with error occurred.
+     *
+     */
+    void Process(const fd_set &aReadFdSet, const fd_set &aWriteFdSet, const fd_set &aErrorFdSet);
+
+    /**
+     * This method returns whether the underlying relay service should append kek after dtls encapsulation
+     * @returns whether the underlying relay service should append kek after dtls encapsulation
+     */
+    bool NeedAppendKek();
```

Comment:
"All void functions or methods shall explicitly declare and specify the void type keyword."